packages/build/src/steps/error.ts

```diff
@@ -81,35 +86,51 @@ const handleFailPlugin = async function ({
   logs,
   debug,
   testOpts,
-}) {
+}: failPluginArgs) {
   const newStatus = serializeErrorStatus({ fullErrorInfo, state: 'failed_plugin' })
   await handleBuildError(newError, { errorMonitor, netlifyConfig, childEnv, mode, logs, debug, testOpts })
-  return { failedPlugin: [packageName], newStatus }
+  // TODO we should probably use type guard here, but due to the way we build these errorInfo objects I'm not 100%
+  // confident we have all the properties currently required by the type
+  const location = fullErrorInfo.errorInfo.location as PluginLocation
+  return { failedPlugin: [location.packageName], newStatus }
```

Comment:
The way we keep adding to the error state makes me reluctant to just use a type guard here.